ledger-core/v2/virtual/execute/execute.go

```diff
@@ -294,23 +294,70 @@ func (s *SMExecute) stepTakeLock(ctx smachine.ExecutionContext) smachine.StateUp
 		return ctx.Sleep().ThenRepeat()
 	}
 
+	if s.Payload.CallRequestFlags.GetRepeatedCall() == payload.RepeatedCall {
+		return ctx.Jump(s.stepDeduplication)
+	}
+
 	return ctx.Jump(s.stepStartRequestProcessing)
 }
 
-func (s *SMExecute) stepStartRequestProcessing(ctx smachine.ExecutionContext) smachine.StateUpdate {
-	var objectDescriptor descriptor.Object
+func (s *SMExecute) stepDeduplication(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	var (
+		isDuplicate       bool
+		pendingListFilled smachine.SyncLink
+	)
+	action := func(state *object.SharedState) {
+		if s.execution.Isolation.Interference == contract.CallIntolerable {
+			pendingListFilled = state.UnorderedPendingListFilled
+		} else {
+			pendingListFilled = state.OrderedPendingListFilled
+		}
+
+		pendingList := state.PendingTable.GetList(s.execution.Isolation.Interference)
+
+		if pendingList.Exist(s.execution.Outgoing) {
+			isDuplicate = true
+			return
+		}
```

Comment:
"This is some serious gourmet" here, you select table based on Interference here, and then get "list" based on interference again. It looks kinda ugly.

Reply:
I get "filled" flag based on Interference and then get pendingList based on Interference. I can do a new structure, that will have them together, will it be ok?

Reply:
Oh, I get it now, misread that part.